ledfx/virtuals.py

```diff
@@ -535,31 +556,58 @@ def flush(self, pixels=None):
         """
         Flushes the provided data to the devices.
         """
+        # _LOGGER.info(f"Virtual {self.id}: Flushing frame")
+
         if pixels is None:
             pixels = self.assembled_frame
+
+        color_cycle = itertools.cycle(color_list)
```

Comment:
Should we try and get this out of the hotpath for non-calibrations? Maybe down to 588?

Reply:
I can put it under the _calibration swtich, it needs to be reset on entry to the start of loop. Its the base iterator for the list, of colors that will be drawn from, rather the color extraction for each segment.

The test will be cheaper than current

Reply:
I can also refactor for speed overall. Will do that now

Reply:
Actually although I can refactor things for speed, there is a risk that the rest api call changes the value of _calibration while an update is in flight, and that can break the look if color_cycle is not substantiated. :-(

As much as I could mutex all this code, I would rather just have it survive...